### PR TITLE
[PDI-12460] LOG_FIELD in Job Log Table entry doesn't capture the enti…

### PR DESCRIPTION
**Pull request by stanislau-strelchanka**
…re execution results - log length of truncated string

### CODE REVIEW

**Comment by stanislau-strelchanka** [comment]
@brosander @mchen-len-son, please review

**Comment by brosander** [comment]
http://wingman.pentaho.com:8080/job/wingman/11/

**Comment by brosander** [comment]
Build Commands:
==============
Build:
------
1. ant -Dtestreports.dir=bin/reports/unit-test -Djunit.jvmargs=-XX:PermSize=256m -f core/build.xml clean-all resolve jacoco && ant -Dpentaho.coding.standards.version=1.0.4 -Dtestreports.dir=bin/reports/integration-test -Djunit.jvmargs=-XX:PermSize=256m -f core/build.xml jacoco-integration checkstyle publish-local

Cleanup:
-------
rm -r ~/.ivy2/local || echo no publish local to remove



Changed files:
==============
  core/src/org/pentaho/di/core/row/value/ValueMetaBase.java



Unit test coverage change:
============
org.pentaho.di.core.logging.LoggingBuffer:
 * Branch Change :  + 1.72%
 * Complexity Change :  + 1.72%
 * Instruction Change :  + 1.45%
 * Line Change :  + 1.53%

org.pentaho.di.core.row.value.ValueMetaBase:
 * Branch Change : **-.05%**
 * Complexity Change : **-.03%**
 * Instruction Change : **-.04%**
 * Line Change : **-.02%**

**Comment by brosander** [comment]
@stanislau-strelchanka this looks like just a little extra logging, is that the agreed upon solution?

**Comment by stanislau-strelchanka** [comment]
@brosander problem is that when setting prepared statement arguments string value is being truncated if it exceeds DatabaseMeta.CLOB_LENGTH (9 999 999 ). This value could be overridden for 
each database meta, but agreed with @mbatchelor that it is better to dump such big values to some file system and add logging of truncated chars.

**Comment by stanislau-strelchanka** [comment]
@brosander, @mattyb149, @mchen-len-son, please review

File: core/src/org/pentaho/di/core/row/value/ValueMetaBase.java [hunk 1]
```diff
@@ -4790,6 +4794,9 @@ public void setPreparedStatementValue( DatabaseMeta databaseMeta, PreparedStatem
               if ( begin < 0 ) {
                 begin = 0;
               }
+              if ( begin > 0 ) {
+                log.logMinimal( String.format( "Truncating %d symbols of original message in '%s' field", begin, getName() ) );
+              }
 
               // Get the substring!
               String logging = string.substring( begin );
```
Comment (mchen-len-son) [inline core/src/org/pentaho/di/core/row/value/ValueMetaBase.java]: once we check that *begin* is positive, let's do extracting _in_ the if block

File: core/src/org/pentaho/di/core/row/value/ValueMetaBase.java [hunk 2]
```diff
@@ -4790,6 +4794,9 @@ public void setPreparedStatementValue( DatabaseMeta databaseMeta, PreparedStatem
               if ( begin < 0 ) {
                 begin = 0;
               }
+              if ( begin > 0 ) {
```
Comment (mchen-len-son) [inline core/src/org/pentaho/di/core/row/value/ValueMetaBase.java]: please, add a small comment why we are doing this

**Comment by stanislau-strelchanka** [comment]
@brosander, @mattyb149, @mchen-len-son, updated, please review

**Comment by brosander** [comment]
http://wingman.pentaho.com:8080/job/wingman/22/

**Comment by brosander** [comment]
Build Commands:
==============
Build:
------
1. ant -Dtestreports.dir=bin/reports/unit-test -Djunit.jvmargs=-XX:PermSize=256m -f core/build.xml clean-all resolve jacoco && ant -Dpentaho.coding.standards.version=1.0.4 -Dtestreports.dir=bin/reports/integration-test -Djunit.jvmargs=-XX:PermSize=256m -f core/build.xml jacoco-integration checkstyle publish-local

Cleanup:
-------
rm -r ~/.ivy2/local || echo no publish local to remove



Changed files:
==============
  core/src/org/pentaho/di/core/row/value/ValueMetaBase.java



Unit test coverage change:
============
org.pentaho.di.core.row.value.ValueMetaBase:
 * Instruction Change : **-.03%**
 * Line Change :  + .03%

**Comment by brosander** [comment]
@stanislau-strelchanka can you add a test for the new functionality please?

**Comment by eliofreitas** [comment]
Rejected from automatic build ,authorized users only

**Comment by stanislau-strelchanka** [comment]
@brosander, @mattyb149, @mchen-len-son, updated, please review

**Comment by lucboudreau** [comment]
http://10.100.11.222:8080/job/wingman/57/

**Comment by lucboudreau** [comment]
Build Commands:
==============
Build:
------
1. ant -Dtestreports.dir=bin/reports/unit-test -Djunit.jvmargs=-XX:PermSize=256m -f core/build.xml clean-all resolve jacoco && ant -Dpentaho.coding.standards.version=1.0.4 -Dtestreports.dir=bin/reports/integration-test -Djunit.jvmargs=-XX:PermSize=256m -f core/build.xml jacoco-integration checkstyle publish-local

Cleanup:
-------
rm -r ~/.ivy2/local || echo no publish local to remove



Changed files:
==============
  core/src/org/pentaho/di/core/row/value/ValueMetaBase.java
  core/test-src/org/pentaho/di/core/row/value/ValueMetaBaseTest.java



Unit test coverage change:
============
org.pentaho.di.core.logging.LogChannel:
 * Branch Change :  + 4.88%
 * Complexity Change :  + 3.23%
 * Instruction Change :  + 3.21%
 * Line Change :  + 2.76%
 * Method Change :  + 2.56%

org.pentaho.di.core.logging.LoggingBuffer:
 * Complexity Change :  + 1.72%
 * Instruction Change :  + 1.09%
 * Line Change :  + 1.53%
 * Method Change :  + 3.45%

org.pentaho.di.core.row.value.ValueMetaBase:
 * Branch Change :  + .48%
 * Complexity Change :  + .20%
 * Instruction Change :  + .57%
 * Line Change :  + .68%

**Pull request merged**